sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/IndexingParameterExtension.java

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.util;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.search.documents.models.IndexingParameters;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Defines extension methods for the IndexingParameters class.
+ */
+public class IndexingParameterExtension {
+    private static final ClientLogger LOGGER = new ClientLogger(IndexingParameterExtension.class);
```

Comment:
Looking at how this class is constructed.. I would expect it to be a builder rather than a collection of static methods that return the object.